dao-voting/tests/harness.rs

```diff
@@ -0,0 +1,1156 @@
+                    deadline: 25,
+                    executed: false,
+                }
+            );
+        }
+    }
+
+    mod revert {
+        use super::*;
+
+        #[tokio::test]
+        #[should_panic]
+        async fn panics_when_duration_is_zero() {
+            let (_gov_token, gov_token_id, deployer, _user, _asset_amount) = setup().await;
+            constructor(&deployer.dao_voting, gov_token_id).await;
+
+            let proposal_transaction = proposal_transaction(gov_token_id);
+            create_proposal(&deployer.dao_voting, 10, 0, proposal_transaction.clone()).await;
+        }
+
+        #[tokio::test]
+        #[should_panic]
+        async fn panics_with_zero_acceptance_percentage() {
+            let (_gov_token, gov_token_id, deployer, _user, _asset_amount) = setup().await;
+            constructor(&deployer.dao_voting, gov_token_id).await;
+
+            let proposal_transaction = proposal_transaction(gov_token_id);
+            create_proposal(&deployer.dao_voting, 0, 10, proposal_transaction.clone()).await;
+        }
+
+        #[tokio::test]
+        #[should_panic]
+        async fn panics_with_over_hundred_acceptance_percentage() {
+            let (_gov_token, gov_token_id, deployer, _user, _asset_amount) = setup().await;
+            constructor(&deployer.dao_voting, gov_token_id).await;
+
+            let proposal_transaction = proposal_transaction(gov_token_id);
+            create_proposal(&deployer.dao_voting, 101, 10, proposal_transaction.clone()).await;
+        }
+    }
+}
+
+mod deposit {
+    use super::*;
+
+    mod success {
+        use super::*;
+
+        #[tokio::test]
+        async fn user_can_deposit() {
+            let (_gov_token, gov_token_id, deployer, user, asset_amount) = setup().await;
+
+            mint(
+                &deployer.gov_token.as_ref().unwrap(),
+                asset_amount,
+                user.wallet.address(),
+            )
+            .await;
+
+            constructor(&deployer.dao_voting, gov_token_id).await;
+
+            assert_eq!(balance(&user.dao_voting).await, 0);
+
+            assert_eq!(
+                user_balance(&user.dao_voting, Identity::Address(user.wallet.address())).await,
+                0
+            );
+
+            let call_params = CallParameters::new(
+                Some(asset_amount),
+                Some(AssetId::from(*gov_token_id)),
+                Some(100_000),
+            );
+            deposit(&user.dao_voting, call_params).await;
+
+            // Make sure that deposit did not erroneously work with 0
+            assert!(asset_amount != 0);
+
+            assert_eq!(balance(&user.dao_voting).await, asset_amount);
+
+            assert_eq!(
+                user_balance(&user.dao_voting, Identity::Address(user.wallet.address())).await,
+                asset_amount
+            );
+        }
+
+        #[tokio::test]
+        async fn user_can_make_multiple_deposits() {
```

Comment:
As you've correctly pointed out this test does not need to exist - plus you've only copy and pasted the code and gave it a different name... delete this test

Reply:
Whoops that was supposed to be deleted...